### PR TITLE
chromium: fix cross compilation

### PR DESCRIPTION
**Pull request by Mic92**
## Description of changes

 - Tested native compilation of chromium with native compilation on x86_64-linux and cross-compilation to aarch64-linux            
 - Before this change we injected cross-compiled libraries into depsDepsBuild inputs.               
   This happens because we override the libpng and libopus package mess                             
   with splicing.                                                                                   

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by emilylange** [comment]
Can you please share details about this?

Your PR description is empty, and your commit message only reads `chromium: fix cross-compilation`.

Cross-compilation puts lots of additional strain on chromium, a derivation that is already in a rough shape and heavily struggling to keep up at all.

How did you test your change?
Which architectures did you test?
Did you test electron, which depends on this chromium derivation, at all?

**Comment by Mic92** [comment]
> Can you please share details about this?
> 
> Your PR description is empty, and your commit message only reads `chromium: fix cross-compilation`.
> 
> Cross-compilation puts lots of additional strain on chromium, a derivation that is already in a rough shape and heavily struggling to keep up at all.
> 
> How did you test your change? Which architectures did you test? Did you test electron, which depends on this chromium derivation, at all?

Updated.

**Comment by avnik** [comment]
@emilylange this should fix #292346 and #292364

**Comment by emilylange** [comment]
I don't see how c4492425adabbe8e75515753590baf1bf51766ca is needed since #298515 has been merged to `staging` and will probably land in `master` the coming weeks.

Additionally, from what I know, `xdg-utils` not cross-compiling does not prevent a chromium/electron cross compilation build from succeeding.

Amjoseph, who is responsible for most of the cross-compilation work in chromium, added `xdg-utils.meta.broken` conditionals for this.

I am fine with the cross-compilation fix for chromium (and by proxy electron) itself.
But also note that I won't notice any issues with it if there are some, because I don't know much about cross-compilation.

Would you mind dropping your `xdg-utils` fix from this PR (because #298515 has been merged instead of this) and rebase and re-target master?

Thanks.

**Comment by Mic92** [comment]
xdg-utils was used in the chromium shell wrapper. It's not used in the chromium build itself. I can check if https://github.com/NixOS/nixpkgs/pull/298515 fixes the issue.

**Comment by emilylange** [comment]
> xdg-utils was used in the chromium shell wrapper. It's not used in the chromium build itself.

Ah, so my information was outdated. `xdg-utils` cross-compilation is no longer marked as broken since #246954. I did not notice that. See #225112 and #225114 for context.

That's good to know.

> I can check if #298515 fixes the issue.

Would be cool if you could report back when you find some time and compute for this :)


> There should be enough engineering capacity in the ghaf project to also ensure in future that cross-compilation will continue to work for chromium.

What is the /graf project/ you are referring to?

**Comment by Mic92** [comment]
It's like QubesOS but using NixOS and KVM-based virtualisation: https://tiiuae.github.io/ghaf/

**Comment by Mic92** [comment]
My xdg-utils fix is no longer needed. Still waiting for the chromium build to finish.

**Comment by Mic92** [comment]
However now I found a mesa regression that is new...

**Comment by Mic92** [comment]
Chromium builds again. Finished from my side.

**Review by Mindavi — APPROVED** [review]

File: pkgs/applications/networking/browsers/chromium/common.nix
Comment (emilylange) [inline pkgs/applications/networking/browsers/chromium/common.nix:1]: If this works and you are available to ping if this breaks again, then this is fine with me.

File: pkgs/by-name/me/meson/package.nix
Comment (emilylange) [inline pkgs/by-name/me/meson/package.nix:1]: #302769 has been merged and thus isn't part of this review (besides, I would be able to comment on it anyway)

**Review by emilylange — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/sphinx/default.nix
[the remark this answers is not in the document]
Comment (Mic92) [reply pkgs/development/python-modules/sphinx/default.nix:1]: We actually use both notations in nixpkgs. It was @FRidh where I saw this way first.

File: pkgs/development/python-modules/sphinx/default.nix [hunk 1]
```diff
@@ -123,6 +123,12 @@ buildPythonPackage rec {
     "test_partialfunction"
   ];
 
+  disabledTestPaths = [
```
[the remark this answers is not in the document]
Comment (Mic92) [reply pkgs/development/python-modules/sphinx/default.nix:126]: There are multiple tests that all break.

File: pkgs/applications/networking/browsers/chromium/common.nix
Comment (Mic92) [reply pkgs/applications/networking/browsers/chromium/common.nix:1]: Sure.

File: pkgs/development/python-modules/sphinx/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mic92) [reply pkgs/development/python-modules/sphinx/default.nix:126]: Done.

File: pkgs/development/python-modules/sphinx/default.nix
[the remark this answers is not in the document]
Comment (Mic92) [reply pkgs/development/python-modules/sphinx/default.nix:1]: Changed.

**Review by emilylange — APPROVED** [review]
Thanks

**Pull request merged**